src/cosalib/qemuvariants.py

```diff
@@ -304,24 +304,31 @@ def mutate_image(self):
             log.info(f"Moving {work_img} to {final_img}")
             shutil.move(work_img, final_img)
 
-        if self.gzip:
-            sha256 = sha256sum_file(final_img)
-            size = os.stat(final_img).st_size
-            temp_path = f"{final_img}.tmp"
-            with open(temp_path, "wb") as fh:
-                runcmd(['gzip', '-9c', final_img], stdout=fh)
-            shutil.move(temp_path, final_img)
-            meta_patch.update({
-                'skip-compression': True,
-                'uncompressed-sha256': sha256,
-                'uncompressed-size': size,
-            })
-
-        if self.skip_compression:
-            meta_patch.update({
-                'skip-compression': True
-            })
-
+        if self.compression is not None:
+            if self.compression == "skip":
+                meta_patch.update({
+                    'skip-compression': True
+                })
+            else:
```

Comment:
nit: This could be `elif self.compression is not None:` and then everything could de-indent by one level.